### PR TITLE
fix: ensure indent-seq works in nested collections

### PR DESCRIPTION
**Pull request by bcbrookman**
Currently, when spaces: consistent, and indent-sequences: true (the defaults), the expected indent for an unindented block sequence is arbitrarily set to 2 expecting it to be at the start of the document. Unfortunately, this causes a bug when the expected indent should be more than 2. For example, in nested collections like the one below, linting would pass, even though the expected indent is _at least_ 3  (since spaces: consistent doesn't enforce how many spaces should be used for indentation). I first reported this behavior in issue #485, which this PR resolves. 

```yaml
---
- item:
  - elem 1
  - elem 2 
```

Instead of setting the expected indent value to 2 arbitrarily, this change sets the value to -1, to explicitly indicate that indentation is required, but the expected indent value is unknown. The problem description that is returned when this wrong indentation is caught then displays the minimum number of spaces expected, since that is known (indent found + 1)

```
3:3       error    wrong indentation: expected at least 3  (indentation)
```

As a side effect, this also fixes a minor related bug I uncovered where the expected indentation printed when linting a nested collection like the one below, is always 2, even though _1 or more_ would also be acceptable with spaces: consistent. 

```yaml
---
item:
- elem 1
- elem 2 
```

### CODE REVIEW

File: tests/rules/test_indentation.py [hunk 1]
```diff
@@ -1369,6 +1369,12 @@ def test_nested_collections(self):
                    '     - - - - - very nested: a\n'
                    '               key: value\n'
                    '...\n', conf, problem=(2, 2))
+        conf = 'indentation: {indent-sequences: true}'
+        self.check('---\n'
+                   '- item:\n'
+                   '  - elem 1\n'
+                   '  - elem 2\n'
+                   '...\n', conf, problem=(3, 3))
```
Comment (adrienverge) [inline tests/rules/test_indentation.py:1377]: Could you move this new test case in a dedicated function, e.g. `test_nested_collections_with_spaces_consistent()`, and add a small link to `https://github.com/adrienverge/yamllint/issues/485` so we easily understand what it's about and avoid touching it too much?

**Review by adrienverge — COMMENTED** [review]
Hello Brian, thanks for working on this!

The change seems to make sense, although I'm not sure of the implications / side effects it could have. Apart from the 2 things it fixes (#485 + the minor related bug you noticed), have you thought about corner cases that could be impacted by this? Could `-1` be misinterpreted somewhere else in the code? Potential changes of behavior for other rare YAML code? (not every possing YAML is covered by our tests unfortunately)

File: tests/rules/test_indentation.py (hunk 1 above)
Comment (bcbrookman) [reply tests/rules/test_indentation.py]: Sure, I just pushed an amended commit with the test moved to its own function with a reference to the issue along with additional tests to include the other possible indent-sequences values.

**Comment by bcbrookman** [comment]
Hi @adrienverge, thank you for reviewing and so sorry it's taken me this long to respond. 

The only corner case I can think of that this might impact is for potential downstream projects that might always expect the specific `wrong indentation: expected %d but found %d` text. This change would add a new possible output `wrong indentation: expected at least %d'`. I expect parsing the output in this way would be very odd and rare, but I could be mistaken. I also didn't find any other places  in the code where `-1` could be misinterpreted because it's only used to check whether the indent found matches the expected indent. 

Thanks again, and I welcome any additional input you or others might have.

**Comment by tamere-allo-peter** [comment]
As far as parsing yamllint's output is concerned, I really think that yamllint should prefix all different warning and error messages with a unique 4 (for example) digits hex number, e.g. Exxxx for errors and Wxxxx for warnings (like what several other linters do already). This way there would be no difficulty, and also yamllint's output could be translated to other languages without breaking any downstream project.

**Review by adrienverge — APPROVED** [review]
@bcbrookman thank you once again! The new commit looks perfect.

About the implication it could have, I agree it would be very odd and rare, so let's take the risk. It should be fairly simple to adapt to the new string, if needed.

@tamere-allo-peter for this purpose, one should use `yamllint -f parsable` which produce an unambiguous output where the line, column, rule name, error level, etc. can be parsed with confidence by any program. In general, such programs are interested by the rule name.
But I believe Brian was mentioning the "human-readable" part of the string (e.g. `wrong indentation: expected at least 2`), which no one should rely on.

**Comment by tamere-allo-peter** [comment]
@adrienverge Yes I know about the parsable format, however this will become ambiguous as soon as i18n comes to yamllint, this is why, IMHO, prefixing each message with a unique ID would be helpful.
